src/archive_writer.rs

```diff
@@ -365,6 +586,89 @@ fn write_symbol_table<W: Write + Seek>(
     )
 }
 
+fn write_symbol_map<W: Write + Seek>(
+    w: &mut W,
+    kind: ArchiveKind,
+    members: &[MemberData],
+    sym_map: &SymMap,
+    members_offset: u64,
+) -> io::Result<()> {
+    let (size, pad) = compute_symbol_map_size_and_pad(members.len(), sym_map);
+    write_symbol_table_header(w, kind, size, 0, 0)?;
+
+    let mut pos: u32 = members_offset.try_into().unwrap();
+
+    w.write_all(&u32::try_from(members.len()).unwrap().to_le_bytes())?;
+    for m in members {
+        w.write_all(&pos.to_le_bytes())?; // member offset
+        pos = pos
+            .checked_add(
+                (m.header.len() + m.data.len() + m.padding.len())
+                    .try_into()
+                    .unwrap(),
+            )
+            .unwrap();
+    }
+
+    w.write_all(&u32::try_from(sym_map.map.len()).unwrap().to_le_bytes())?;
+
+    for s in sym_map.map.values() {
+        w.write_all(&s.to_le_bytes())?;
+    }
+    for s in sym_map.map.keys() {
+        w.write_all(s)?;
+        w.write_all(&[0])?;
+    }
+
+    write!(
+        w,
+        "{nil:\0<pad$}",
+        nil = "",
+        pad = usize::try_from(pad).unwrap()
+    )?;
+    Ok(())
+}
+
+fn write_ec_symbols<W: Write + Seek>(w: &mut W, sym_map: &SymMap) -> io::Result<()> {
+    let (size, pad) = compute_ec_symbols_size_and_pad(sym_map);
+    print_gnu_small_member_header(w, "/<ECSYMBOLS>".to_string(), now(), 0, 0, 0, size)?;
+
+    w.write_all(&u32::try_from(sym_map.ec_map.len()).unwrap().to_le_bytes())?;
+
+    for s in sym_map.ec_map.values() {
+        w.write_all(&s.to_le_bytes())?;
+    }
+    for s in sym_map.ec_map.keys() {
+        w.write_all(s)?;
+        w.write_all(&[0])?;
+    }
+
+    write!(
+        w,
+        "{nil:\0<pad$}",
+        nil = "",
+        pad = usize::try_from(pad).unwrap()
+    )?;
+    Ok(())
+}
+
+fn is_ec_object(obj: &SymbolicFile) -> bool {
+    match &obj.parsed {
+        File::Coff(coff_obj) => {
+            coff_obj.architecture() != Architecture::Aarch64
+                || coff_obj.sub_architecture() == Some(SubArchitecture::Arm64EC)
```

Comment:
Does this cover import files?

Reply:
Added in support for import files, but not for bitcode - we can either try to add this later, or have the LLVM backend provide an implementation that does check for bitcode.

Reply:
Support for reading bitcode is currently done by the LLVM backend of rustc by overriding `get_symbols` (now `object_reader`).